doc/src/devdocs/inference.md

```diff
@@ -96,7 +96,7 @@ Each statement gets analyzed for its total cost in a function called
 as follows:
 ```jldoctest; filter=r"tuple.jl:\d+"
 julia> Base.print_statement_costs(stdout, map, (typeof(sqrt), Tuple{Int},)) # map(sqrt, (2,))
-map(f, t::Tuple{Any}) in Base at tuple.jl:179
+map(f::Any, t::Tuple{Any}) @ Base tuple.jl:218
```

Comment:
this seems like a regression

Reply:
Can you explain what you mean? 

I just landed here by chasing errors, and have not looked for the code which generates this. I believe that `Base.Math.throw_complex_domainerror(:sqrt::Symbol, %2::Float64)::Union{}` just below in this output does not take the path this PR changes, since it does not get printed in bold:

<img width="763" alt="Screenshot 2022-01-19 at 18 09 22" src="https://user-images.githubusercontent.com/32575566/150233030-97381375-8ff2-4488-bc5e-a3f2fa99a65b.png">

Is there an alternative path that the first line's printing should take?

Reply:
The added `::Any` is unnecessary, and seems to be explicitly being added in this PR

Reply:
Ok, I can surely figure out a way to remove it here. 

But the logic of including it in general is this:

https://user-images.githubusercontent.com/32575566/119697648-6b188f00-be1e-11eb-91e3-c7ae8612f2e4.png

On the right you can count the arguments by counting the bold words. On the left the grey `f,` is really easy to miss, and seems worse than before.